lib/src/slices/data_bkg/data_bkg_service.dart

```diff
@@ -208,4 +209,29 @@ class DataBkgService extends ChangeNotifier {
         "." +
         periodSplit[periodSplit.length - 1];
   }
+
+  String _gmailBuildQuery(int appDataGmailLastFetch, String category) {
+    StringBuffer queryBuffer = new StringBuffer();
+    if (appDataGmailLastFetch > 0) {
+      DateTime gmailLastFetch =
+          DateTime.fromMillisecondsSinceEpoch(appDataGmailLastFetch);
+      int secondsSinceEpoch =
+          (gmailLastFetch.millisecondsSinceEpoch / 1000).floor();
+      queryBuffer.write("after:" + secondsSinceEpoch.toString());
+    }
+    if (category != 'category:' && category.isNotEmpty) {
+      if (queryBuffer.isNotEmpty) {
```

Comment:
I like this design of using the queryBuffer.isNotEmpty to add the " AND " statement. A nice design would be to break the query builder into 2 functions. 1 called add query that does the appending  and the current one just builds the queries and calls the add function